cocotb/config.py

```diff
@@ -38,6 +38,15 @@
     share_dir: str, path where the cocotb data is stored
     makefiles_dir: str, path where the cocotb makefiles are installed
 """
+
+# The output (on STDOUT) of cocotb-config is typically assigned to variables in
+# our makefiles. All additional output breaks this use case and must be avoided
+# if possible. DeprecationWarnings by third-party libraries are not something we
+# can control; not showing those avoids breaking cocotb-config for our users as
+# they make changes to the list of installed Python packages.
+import warnings
+warnings.filterwarnings("ignore", category=DeprecationWarning)
+
```

Comment:
I find it hard to believe this makes any difference. The error in private CI only starts appearing when we change to doing `pip install -e`.

The warning is being emitted by the virtualenv code that launches `cocotb.config`, so is beyond our control.

Reply:
I'll see if I can show you exactly what that wrapper code looks like...

Reply:
Nope can't find it. But if you look at the tracebacks that I mentioned before, not one line referenced any of our files.